controllers/src/main/scala/cromwell/pipeline/controller/AkkaHttpClient.scala

```diff
@@ -0,0 +1,85 @@
+package cromwell.pipeline.controller
+
+import java.nio.charset.Charset
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.http.scaladsl.model.{ HttpMethods, HttpRequest, HttpResponse }
+import akka.http.scaladsl.{ Http, HttpExt }
+import akka.stream.ActorMaterializer
+import akka.util.ByteString
+import cromwell.pipeline.service.{ HttpClient, Response }
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+class AkkaHttpClient extends HttpClient {
+  private implicit val actorSystem: ActorSystem = ActorSystem()
+  private implicit val materializer: ActorMaterializer = ActorMaterializer()
+
+  def http: HttpExt = Http()
+
+  override def get(url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map())(
+    implicit ec: ExecutionContext
+  ): Future[Response] = {
+    val futureResponse: Future[HttpResponse] = http.singleRequest(
+      HttpRequest(method = HttpMethods.GET, uri = queryBuilder(url, params)).withHeaders(parseHeaders(headers))
+    )
+    responsify(futureResponse)
+  }
+
+  override def post(
+    url: String,
+    params: Map[String, String] = Map(),
+    headers: Map[String, String] = Map(),
+    payload: String = ""
+  )(implicit ec: ExecutionContext): Future[Response] = {
+    val futureResponse: Future[HttpResponse] = http.singleRequest(
+      HttpRequest(method = HttpMethods.POST, uri = queryBuilder(url, params))
+        .withEntity(payload)
+        .withHeaders(parseHeaders(headers))
+    )
+    responsify(futureResponse)
+  }
+
+  override def put(
+    url: String,
+    params: Map[String, String] = Map(),
+    headers: Map[String, String] = Map(),
+    payload: String = ""
+  )(implicit ec: ExecutionContext): Future[Response] = {
+    val futureResponse: Future[HttpResponse] = http.singleRequest(
+      HttpRequest(method = HttpMethods.PUT, uri = queryBuilder(url, params))
+        .withEntity(payload)
+        .withHeaders(parseHeaders(headers))
+    )
+    responsify(futureResponse)
+  }
+
+  private def responsify(futureResponse: Future[HttpResponse])(implicit ec: ExecutionContext): Future[Response] =
+    futureResponse.flatMap { response: HttpResponse =>
+      response.entity.dataBytes.runFold(ByteString(""))(_ ++ _).map { body =>
```

Comment:
I'm pretty sure this is unnecessary, in this case we can use just `response.entity.strict(timeout)`.

Reply:
Change it, thanks